grpc/public/passwordless.go

```diff
@@ -0,0 +1,73 @@
+package public
+
+import (
+	"github.com/keratin/authn-server/app"
+	"github.com/keratin/authn-server/app/services"
+	authnpb "github.com/keratin/authn-server/grpc"
+	"github.com/keratin/authn-server/grpc/internal/errors"
+	"github.com/keratin/authn-server/grpc/internal/meta"
+	context "golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+)
+
+type passwordlessServer struct {
+	app *app.App
+}
+
+var _ authnpb.PasswordlessServiceServer = passwordlessServer{}
+
+func (s passwordlessServer) RequestPasswordlessLogin(ctx context.Context, req *authnpb.RequestPasswordlessLoginRequest) (*authnpb.RequestPasswordlessLoginResponse, error) {
+
+	account, err := s.app.AccountStore.FindByUsername(req.GetUsername())
+	if err != nil {
+		panic(err)
+	}
+
+	// run in the background so that a timing attack can't enumerate usernames
+	go func() {
+		err := services.PasswordlessTokenSender(s.app.Config, account)
+		if err != nil {
+			s.app.Reporter.ReportError(err)
+		}
+	}()
+
+	return &authnpb.RequestPasswordlessLoginResponse{}, nil
+}
+
+func (s passwordlessServer) SubmitPasswordlessLogin(ctx context.Context, req *authnpb.SubmitPasswordlessLoginRequest) (*authnpb.SubmitPasswordlessLoginResponseEnvelope, error) {
+
+	var err error
+	var accountID int
+
+	accountID, err = services.PasswordlessTokenVerifier(
+		s.app.AccountStore,
+		s.app.Reporter,
+		s.app.Config,
+		req.GetToken(),
+	)
+
+	if err != nil {
+		if fe, ok := err.(services.FieldErrors); ok {
+			return nil, errors.ToStatusErrorWithDetails(fe, codes.FailedPrecondition).Err()
+		}
+		panic(err)
```

Comment:
As another example, this looks like a sanity panic, i.e. if we get anything other then `FieldErrors` we panic - really?